wrappers/IWearActuators/src/IWearActuatorsWrapper.cpp

```diff
@@ -0,0 +1,221 @@
+public:
+    std::string attachedWearableDeviceName;
+
+    std::string actuatorCommandInputPortName;
+    yarp::os::BufferedPort<wearable::msg::WearableActuatorCommand> actuatorCommandInputPort;
+
+    std::unordered_map<std::string, wearable::DevicePtr<const actuator::IActuator>> actuatorsMap;
+
+    msg::WearableActuatorCommand wearableActuatorCommand;
+
+    wearable::IWear* iWear = nullptr;
+};
+
+IWearActuatorsWrapper::IWearActuatorsWrapper()
+    : yarp::os::PeriodicThread(DefaultPeriod)
+    , pImpl{new impl()}
+{}
+
+IWearActuatorsWrapper::~IWearActuatorsWrapper()
+{
+    detachAll();
+    close();
+}
+
+void IWearActuatorsWrapper::run()
+{
+    //TODO: Send the values of the actuators state ?
+}
+
+// ======================
+// DeviceDriver interface
+// ======================
+
+bool IWearActuatorsWrapper::open(yarp::os::Searchable& config)
+{
+    if (!config.check("actuatorCommandInputPortName") || !config.find("actuatorCommandInputPortName").isString()) {
+        yError() << LogPrefix << "actuatorCommandInputPortName parameter not found";
+        return false;
+    }
+
+
+    if (!config.check("period")) {
+        yInfo() << LogPrefix << "Using default period: " << DefaultPeriod << "s";
+    }
+
+    // Parse configuration parameters
+
+    const double period = config.check("period", yarp::os::Value(DefaultPeriod)).asFloat64();
+    setPeriod(period);
+
+    pImpl->actuatorCommandInputPortName = config.find("actuatorCommandInputPortName").asString();
+
+
+    // Configure yarp ports
+
+    if(!pImpl->actuatorCommandInputPort.open(pImpl->actuatorCommandInputPortName))
+    {
+        yError() << "Failed to open " << pImpl->actuatorCommandInputPortName << " yarp port";
+        return false;
+    }
+
+    // Set the callback to use onRead() method of this device
+    pImpl->actuatorCommandInputPort.useCallback(*this);
+
+    return true;
+}
+
+void IWearActuatorsWrapper::onRead(msg::WearableActuatorCommand& wearableActuatorCommand)
+{
+   // Unpack the actuator in from incoming command
+   wearable::msg::ActuatorInfo info = wearableActuatorCommand.info;
+
+   // Check if the commanded actuator name is available
+   if (pImpl->actuatorsMap.find(info.name) == pImpl->actuatorsMap.end())
+   {
+       yError() << "Requested actuator with name " << info.name << " is not available in " << pImpl->attachedWearableDeviceName << " wearable device!";
+   }
+
+   wearable::actuator::ActuatorType aType = pImpl->actuatorsMap[info.name]->getActuatorType();
+   switch (aType) {
+    case wearable::actuator::ActuatorType::Motor: {
+
+       // Check if the actuator type in the command is correct
+       // TODO: May be this is redundant check ?
+       if (info.type == wearable::msg::ActuatorType::MOTOR)
+       {
+           // Sent the command value to the motor actuator
+            wearable::DevicePtr<const wearable::actuator::IMotor> castActuator = std::dynamic_pointer_cast<const wearable::actuator::IMotor>(pImpl->actuatorsMap[info.name]);
```

Comment:
Dynamic cast called in a loop is not good for performance. If you're sure that the cast is always correct, use static cast.

Reply:
Done https://github.com/robotology/wearables/pull/105/commits/e772a1becdeba13a7c206cffce70c570469126cc